### ISSUE TITLE
migrate issue links in source to moved issues

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4461]
I admit I don't totally know what's involved here so I'd very much welcome insights. The rub is in source we've got a handful of issue references that will need to be evaluated and updated when we migrate issues to the SDK tracker.  Some are probably stale and can reasonably be deleted but in at least the case of failing tests annotated like this:

```dart
  @FailingTest(issue: 'https://github.com/dart-lang/linter/issues/3516')
  test_matchingArg_dynamicParameterToOtherArg_positional() async { ... }
```

I would like to preserve the context of linked issues.

@srawlins: not sure if this is an issue for dartdoc, but FYI.

Also fyi @parlough, in case you have any thoughts.

**Comment by srawlins** [comment]
Yeah we should just use GitHub's "Transfer issue", available with the GitHub commandline tool.

Old issue URLs are preserved as redirects to the new ones, so we don't have to clean up references. We can but we don't have to.

**Comment by srawlins** [comment]
This script looks good for bulk transfering: https://jloh.co/posts/bulk-migrate-issues-github-cli/

**Comment by pq** [comment]
Awesome! Hopefully these redirects will work even when a repo is archived? 🤞

**Comment by whesse** [comment]
I modified all the issue links in the linter repo to read dart-lang/linter#1234 instead of #1234, and uploaded it to https://github.com/whesse/linter .  There were 2595 cases where I added dart-lang/linter, and 49 cases where the issue reference was already to a different repo.  Take a look and see if it looks good to you.

**Comment by pq** [comment]
LGTM! (fyi @stereotype441)
